File: backend/src/main/java/pl/edu/ur/blokur/config/FlywayConfig.java
package pl.edu.ur.blokur.config;

import javax.sql.DataSource;
import org.flywaydb.core.Flyway;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

/**
 * Konfiguracja narzędzia Flyway. Ręcznie inicjuje proces migracji bazy danych przy starcie
 * aplikacji, wymuszając wykonanie skryptów.
 */
@Configuration
public class FlywayConfig {

    @Value("${spring.flyway.locations:classpath:db/migration}")
    private String[] locations;

    @Bean(initMethod = "migrate")
    public Flyway flyway(DataSource dataSource) {
        Flyway flyway =
                Flyway.configure()
                        .dataSource(dataSource)
                        .locations(locations)
                        .baselineOnMigrate(true)
                        .load();
        return flyway;
    }
}

File: backend/src/main/kotlin/pl/edu/ur/blokur/config/FlywayConfig.kt
package pl.edu.ur.blokur.config

import jakarta.annotation.PostConstruct
import org.flywaydb.core.Flyway
import org.springframework.context.annotation.Configuration
import javax.sql.DataSource

@Configuration
class FlywayConfig(
    private val dataSource: DataSource,
) {
    @PostConstruct
    fun migrateFlyway() {
        try {
            val flyway =
                Flyway
                    .configure()
                    .dataSource(dataSource)
                    .locations("classpath:db/migration")
                    .load()
            flyway.migrate()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}
